packages/amplify-category-auth/src/provider-utils/awscloudformation/index.ts

```diff
@@ -218,12 +226,13 @@ const getHeadlessParams = context => {
   }
 };
 
-const getOAuthProviderKeys = (currentEnvSpecificValues, resourceParams) => {
-  const oAuthProviders = JSON.parse(resourceParams.hostedUIProviderMeta).map(h => h.ProviderName);
+/* eslint-disable no-param-reassign */
```

Comment:
I think we should disable this rule.

Reply:
Broadly, I don't know that I agree that no-param-reassign should be disabled. I disabled it here just because the code modifies the parameter in this block, and the objective is to make minimal changes for the migration from JS to TS, and refactoring this to not reassign the parameter feels out of scope.

Reply:
I have mixed feelings about this particular rule, because it doesn't always lead to more readable code.

I think the rule is useful in code blocks that span across a lot of lines (i.e. a screen or more), but on the other hand one should not write code like that.
In smaller blocks defining new var/const just for the sake of satisfying this rule often leads to weird naming, especially if it's series of transforms.

Besides, regardless of how do we like this rule we seem to use it quite often, see https://github.com/aws-amplify/amplify-cli/search?q=no-param-reassign .

Let's talk about this when we trim rulset.